### PR TITLE
PT-1.11 CVE patching

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ShiboXing** [comment]
standard: aacde98
rc: 7402d84
efa: a685c61

File: pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu [hunk 1]
```diff
@@ -347,6 +347,8 @@ RUN HOME_DIR=/root \
  && rm -rf ${HOME_DIR}/oss_compliance* \
  && rm -rf /tmp/tmp*
 
+RUN rm -rf ~/.cache
```
Comment (kace) [inline pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu:350]: What's the reason for adding this?

File: pytorch/training/docker/1.11/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -296,7 +298,7 @@ RUN pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pytho
     smclarify \
     "sagemaker>=2,<3" \
     sagemaker-experiments==0.* \
-    "sagemaker-pytorch-training>=2.5.1,<3"
```
Comment (kace) [inline pytorch/training/docker/1.11/py3/Dockerfile.cpu:299]: Do we know why this was pinned originally?

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def test_dgl_gcn_training_gpu(ecr_image, sagemaker_regions, instance_type):
         "instance_type": instance_type,
     }
 
-    if Version(image_framework_version) in SpecifierSet(">=1.12"):
+    if Version(image_framework_version) in SpecifierSet(">=1.11.0"):
```
Comment (kace) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py:74]: What changed here?

Comment (ShiboXing) [reply test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py:74]: We haven't pinned the dgl package and the latest got updated to 0.9.1, hence the change

File: pytorch/training/docker/1.11/py3/Dockerfile.cpu (hunk 2 above)
Comment (ShiboXing) [reply pytorch/training/docker/1.11/py3/Dockerfile.cpu:299]: I'm not sure, but the to consume latest protobuf deprecation fix and the hopper fix, we need to upgrade.

File: pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu (hunk 1 above)
Comment (kace) [reply pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu:350]: Got it. So this is a backport of something we've implemented in images with newer FW versions.

File: pytorch/training/docker/1.11/py3/Dockerfile.cpu (hunk 2 above)
Comment (kace) [reply pytorch/training/docker/1.11/py3/Dockerfile.cpu:299]: Sounds good.

**Review by kace — APPROVED** [review]

**Pull request merged**